test/cli-doctor-stop-delay.test.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+const fs = require('fs')
+const url = require('url')
+const async = require('async')
+const path = require('path')
+const test = require('tap').test
+const cli = require('./cli.js')
+
+test('clinic doctor --stop-delay --on-port - no issues', function (t) {
+  cli({}, [
+    'clinic', 'doctor', '--no-open', '--stop-delay', '2000', '--on-port', 'node -e "setTimeout(() => {}, 1000)"',
+    '--', 'node', '-e', `
+      const http = require('http')
+
+      http.createServer((req, res) => res.end('ok')).listen(0)
+    `
+  ], function (err, stdout, stderr, tempdir) {
+    t.error(err)
+
+    const dirname = stdout.match(/(\.clinic[/\\]\d+.clinic-doctor)/)[1]
+    const fullpath = url.pathToFileURL(fs.realpathSync(path.resolve(tempdir, dirname)))
+    t.equal(stdout.split('\n')[0], 'Waiting to close the process')
+    t.equal(stdout.split('\n')[1], 'Analysing data')
+    t.equal(stdout.split('\n')[2], `Generated HTML file is ${fullpath}.html`)
+
+    // check that files exists
+    async.parallel({
+      sourceData (done) {
+        fs.access(path.resolve(tempdir, dirname), done)
+      },
+      htmlFile (done) {
+        fs.access(path.resolve(tempdir, dirname + '.html'), done)
+      }
+    }, function (err) {
+      t.error(err)
+      t.end()
+    })
+  })
+})
+
+// TODO: skipping until the above statement is fulfilled
+test('clinic doctor --stop-delay --on-port - exceeding timeout', { skip: true }, function (t) {
+  // TODO: rafaelgss
```

Comment:
@mcollina any idea how to make it work?

Reply:
why it's not working?

Reply:
`node-tap` doesn't have an API for an expected timeout. When a `timeout` is raised it automatically mark the test as `not ok`.

Reply:
What's wrong in using setTimeout?